### PR TITLE
Make reformatting options more verbose

### PR DESCRIPTION
**Pull request by ajaust**
## Main changes of this PR

- Mention `reformat-all` script in PR template
- `check-format` mentions `reformat-all` script if some files are formatted wrongly.
 
  ```bash
  $ tools/formatting/check-format
  Using binary: clang-format-8
  GNU parallel detected.
  Checking 388 C/C++ files
  100% 388:0=0s tools/testing/communication_dummies/mainB.cpp                                                                                 
  Checking 100 XML files
  100% 100:0=0s src/xml/tests/xmltest-unique-attribute-config.xml                                                                             
  The following files are not formatted correctly:
  src/precice/impl/SolverInterfaceImpl.cpp
  You can automatically reformat all files using the script tools/formatting/format-all
  ```

  Note, that the last line in the output is new.

## Motivation and additional information

I found author's checklist of limited help when it comes to format checking. I ran the `check-format` tool and my files were formatted wrongly, see

```bash
$ tools/formatting/check-format
Using binary: clang-format-8
GNU parallel detected.
Checking 388 C/C++ files
100% 388:0=0s tools/testing/communication_dummies/mainB.cpp                                                                                 
Checking 100 XML files
100% 100:0=0s src/xml/tests/xmltest-unique-attribute-config.xml                                                                             
The following files are not formatted correctly:
src/precice/SolverInterface.hpp
src/precice/impl/SolverInterfaceImpl.hpp
```

However, it did not give me insight regarding what is wrong, nor a patch to apply, nor any hint to the `format-all` script. Sadly, the `README.md` in `tools/formatting` does not help much either.

## Author's checklist

* [ ] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
* [x] I ran `tools/formatting/check-format` and everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

**Comment by ajaust** [comment]
This is a first suggestion on how help contributors to format the code properly. One could also try to solve this by more extensive documentation in the README.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Short rational why preCICE needs this change. If this is already described in an
 ## Author's checklist
 
 * [ ] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
-* [ ] I ran `tools/formatting/check-format` and everything is formatted correctly.
+* [ ] I ran `tools/formatting/check-format` and everything is formatted correctly. Run `tools/formatting/format-all` to reformat files automatically.
```
Comment (fsimonis) [inline .github/pull_request_template.md:13]:
```suggestion
* [ ] I ran `tools/formatting/format-all` to ensure everything is formatted correctly.
```

**Comment by fsimonis** [comment]
Actually, it is way simpler/more efficient way is to simply run `format-all` instead of `check-format` :sweat_smile:

File: .github/pull_request_template.md (hunk 1 above)
Comment (ajaust) [reply .github/pull_request_template.md:13]: I was not sure if one suggests this, without any risk of messing something up. I never had the reformatting fail for me, but who knows. I also prefer the shorter/simpler solution. :smile:

**Comment by fsimonis** [comment]
I guess the simplest way would be to integrate this into CMake. Then you don't even have to bother with the prefix.
```
$ make format
```

**Pull request merged**